File: android/src/main/java/fr/g123k/deviceapps/AsyncWork.java
package fr.g123k.deviceapps;

import java.util.concurrent.BlockingQueue;
import java.util.concurrent.LinkedBlockingQueue;
import java.util.concurrent.ThreadPoolExecutor;
import java.util.concurrent.TimeUnit;

class AsyncWork {

    private final ThreadPoolExecutor threadPoolExecutor;
    private final BlockingQueue<Runnable> workQueue;

    AsyncWork() {
        this.workQueue = new LinkedBlockingQueue<>();
        this.threadPoolExecutor = new ThreadPoolExecutor(1, 1, 1, TimeUnit.SECONDS, workQueue);
    }

    void run(Runnable runnable) {
        threadPoolExecutor.execute(runnable);
    }

    void stop() {
        threadPoolExecutor.shutdown();
    }
}

File: android/src/main/java/fr/g123k/deviceapps/AsyncWork.kt
package fr.g123k.deviceapps

import java.util.concurrent.LinkedBlockingQueue
import java.util.concurrent.ThreadPoolExecutor
import java.util.concurrent.TimeUnit

internal class AsyncWork {
    private val threadPoolExecutor: ThreadPoolExecutor  = ThreadPoolExecutor(1,
            1, 1,
            TimeUnit.SECONDS, LinkedBlockingQueue())
    fun run(runnable: Runnable?) = threadPoolExecutor.execute(runnable)

    fun stop() = threadPoolExecutor.shutdown()
}